src/YoutubeAtom.tsx

```diff
@@ -227,13 +227,20 @@ export const YoutubeAtom = ({
                       adTargeting.customParams,
                       consentState,
                   );
-        const embedConfig = encodeURIComponent(JSON.stringify({ adsConfig }));
-        const originString = origin
-            ? `&origin=${encodeURIComponent(origin)}`
-            : '';
-        setIframeSrc(
-            `https://www.youtube.com/embed/${assetId}?embed_config=${embedConfig}&enablejsapi=1&widgetid=1&modestbranding=1${originString}&autoplay=1`,
+
+        const params = new URLSearchParams({
+            ...(origin ? { origin } : {}),
```

Comment:
```suggestion
            ...(origin ?? {}),
```

Reply:
Wait this doesn't work! Never mind.

Reply:
I think this won’t lead to our expected result

```ts
const origin = "https://www.example.com/";
console.log({ ...{ origin : origin } }); // { origin: "https://www.example.com/" }
// if you have a const in scope that matches a key, you can omit it in the below syntax, which is what I’m doing
console.log({ ...{ origin } }); // { origin: "https://www.example.com/" }

console.log({ ...origin }); // { "0": "h", "1": "t", "2": "t", "1": "p", … }
```